docs/developer/designs/topology-awareness/README.md

```diff
@@ -0,0 +1,306 @@
+# Topology Aware Scheduling
+
+## Summary
+
+This design document outlines an approach for implementing topology aware scheduling in KAI-Scheduler, leveraging the topology CRD concepts from Kueue. Topology awareness enables more efficient scheduling of workloads by considering the physical placement of resources within a cluster, leading to improved performance for applications that benefit from resource locality.
+
+## Motivation
+
+Workloads with specific topology requirements, such as distributed training jobs or applications sensitive to inter-node communication latency, benefit significantly from intelligent placement decisions that respect the underlying hardware topology. Without topology awareness, jobs may be scheduled across physically distant nodes, leading to:
+
+- Increased network latency and reduced bandwidth
+- Suboptimal performance for compute-intensive workloads
+- Inefficient resource utilization across the cluster
+
+## Usage Stories
+
+### Better Inter Pod Communication
+
+Distributed workload with high volume of communication between pods will prefer the pods to be as close as possible, with the highest volume and speed of network between them.
+If not possible to allocate them on the same node, then on the same rack or block. It is more important that it starts running than that it fills the exact topology requirements.
+
+### Highly Optimized With Regard to Network Infrastructure
+
+This workload wants to make sure that it takes advantage of the best setup its network can give.
+It wants to split the pods into groups that will fill entire nodes, and the groups be on the same rack if possible, or close one from the same block.
+The pods then need to get some hints to their respective location in this internal job topology.
+
+## Goals
+
+- Implement topology-aware scheduling in KAI-Scheduler
+- Leverage Kueue's topology CRD concepts to keep an industry standard
+   - Support hierarchical topology representations (e.g., datacenter/rack/node/device)
+- Provide flexible configuration options for topology-based placement
+- Support both Required and Preferred topology constraint request
+- Checking more possible allocations than is possible for PodAffinity pod groups: When the scheduler tries to scheudler a pod group with multiple pods that have a pod affinity requirement it finds a suitable node for the  first pods and that node's label is then used to compare the pod affinity of the next pods in the pod group. If the rest of the pods can't be allocated with that node affinity then the first pod's allocation is not attempted on another node. We aim to fix this issue
+
+## Non-Goals
+
+- Automatically detecting topology without user input
+- Ensure backward compatibility with existing scheduling mechanisms
```

Comment:
I still don't understand the second point, now that you specifically mentioned that TAS is going to be a better alternative for PadAffinity - what existing scheduling mechanisms are not going to supported with TAS?

Reply:
Can we say exactly that so it will not sound as if there is no BC here?